src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import {
+  Popover as CorePopover,
+  PopoverProps as CorePopoverProps,
+} from 'wix-ui-core/popover';
+
+import { TPAComponentProps } from '../../types';
+import { generateKey } from '../../common/random';
+import { KEYS } from '../../common/keyCodes';
+
+import { TriggerAction } from './constants';
+import { st, classes } from './Popover.st.css';
+
+export interface PopoverProps
+  extends TPAComponentProps,
+    Omit<CorePopoverProps, 'shown'> {
+  triggerAction?: TriggerAction;
+  wiredToSiteColors?: boolean;
+}
+
+export const Popover: React.FC<PopoverProps> & {
+  Element?: React.FC;
+  Content?: React.FC;
+} = props => {
+  const {
+    triggerAction,
+    wiredToSiteColors,
+    children,
+    className,
+    ...rest
+  } = props;
+  const [isShown, setIsShown] = React.useState<boolean>(false);
+  const [id] = React.useState(generateKey('popover'));
```

Comment:
In this case it's better to use `useRef`
```suggestion
  const id = React.useRef(generateKey('popover'));
```
But are hooks allowed in the platform?